IBPSA/Fluid/Chillers/BaseClasses/InnerCycle_Chiller.mo

```diff
@@ -0,0 +1,209 @@
+within IBPSA.Fluid.Chillers.BaseClasses;
+model InnerCycle_Chiller "Blackbox model of refrigerant cycle of a chiller"
+  extends IBPSA.Fluid.HeatPumps.BaseClasses.PartialInnerCycle;
+  // Setting all values to zero avoids errors when checking this model.
+  // The values are correctly propagated by the heat pump / chiller model anyway
+  replaceable model BlaBoxChiCooling =
+      IBPSA.Fluid.Chillers.BlackBoxData.BaseClasses.PartialChillerBlackBox (
+       QUse_flow_nominal=0,
+       QUseBlackBox_flow_nominal=0,
+       scalingFactor=0,
+       TCon_nominal=0,
+       TEva_nominal=0,
+       dTCon_nominal=0,
+       dTEva_nominal=0,
+       primaryOperation=true,
+       mCon_flow_nominal=0,
+       mEva_flow_nominal=0,
+       y_nominal=0)
+    constrainedby
+    IBPSA.Fluid.Chillers.BlackBoxData.BaseClasses.PartialChillerBlackBox
+    "Replaceable model for performance data of a chiller in main operation mode"
+    annotation (choicesAllMatching=true);
+
+  replaceable model BlaBoxChiHeating =
+      IBPSA.Fluid.HeatPumps.BlackBoxData.BaseClasses.NoHeating(primaryOperation=false)
+    constrainedby
+    IBPSA.Fluid.HeatPumps.BlackBoxData.BaseClasses.PartialHeatPumpBlackBox
+    "Replaceable model for performance data of a chiller in reversible operation mode"
+    annotation (Dialog(enable=use_rev),choicesAllMatching=true);
+
+  BlaBoxChiCooling BlackBoxChillerCooling
+    annotation (Placement(transformation(extent={{7,20},{61,76}}, rotation=0)));
+  BlaBoxChiHeating BlackBoxChillerHeating if use_rev
+    annotation (Placement(
+        transformation(
+        extent={{-27,-28},{27,28}},
+        rotation=0,
+        origin={-34,48})));
+
+  Modelica.Blocks.Math.Gain gainEva(final k=-1)
+    "Negate QEva to match definition of heat flow direction" annotation (
+      Placement(transformation(
+        extent={{-4,-4},{4,4}},
+        rotation=180,
+        origin={-50,-6})));
+  Modelica.Blocks.Math.Gain gainCon(final k=-1) if use_rev
+    "Negate QCon to match definition of heat flow direction" annotation (
+      Placement(transformation(
+        extent={{-4,-4},{4,4}},
+        rotation=0,
+        origin={50,-20})));
+protected
+  Utilities.IO.Strings.StringPassThrough stringPassThrough;
+  Utilities.IO.Strings.ConstStringSource constStringSource(final k=BlackBoxChillerCooling.datasource) if not use_rev;
```

Comment:
Sadly, it does not work in Dymola 2022x. I will try with 2023 asap. 
One point on the rule using 3 letters only. Even tough it's a convention, following this blindy results in less readable code. Using `datSou`, users don't know what it means and have to look up the documentation. Using `datasource`, it's clearer. Same applies to `scalingFactor`, `primaryOption` and similar. I can apply the rule in all these cases, but do you never have exceptions, except the standard (allowFlowReversal, etc.)?

Reply:
Yes, we had long discussions many years ago and settled on the 3 letter camel case as a compromise between short names and being able to understand what they are. The comment string of the parameter explains that `scaFac` is a scaling factor. Using the 3 letter convention *consistently* makes it much easier to browse through results, search the result tree for variables etc.

Reply:
Ok! I renamed all variables accordingly.

Reply:
Also to add to this issue, I have found it relatively common to break this rule and use full words for Boolean switches, such as `allowFlowReversal` in `IBPSA.Fluid.Interfaces.PartialTwoPort` and `computePowerUsingSimilarityLaws` (among others) in `IBPSA.Fluid.Movers.BaseClasses.PartialFlowMachine`. But I don't know whether it is because these models predate the establishment of the naming convention though.